src/cursor/cur_table.c

```diff
@@ -972,7 +972,8 @@ __wt_curtable_open(WT_SESSION_IMPL *session,
 
 	if (0) {
 err:		if (*cursorp != NULL) {
-			WT_TRET(__wt_cursor_close(*cursorp));
+			if (*cursorp != cursor)
+				WT_TRET(__wt_cursor_close(*cursorp));
```

Comment:
Yes, the only time on any call to `wt_cursor_init` that they're not equal is when opening a dump cursor.